draco/analysis/flagging.py

```diff
@@ -606,27 +605,25 @@ def process(self, data):
             weights substituted by the smoothed ones.
         """
 
-        # Ensure data is distributed in frequency:
+        # Ensure data is distributed in frequency,
+        # so a frequency loop will not be too large.
         data.redistribute("freq")
-        # Full slice reutrns an MPIArray
-        weight = data.weight[:]
-        # Data will be distributed in frequency.
-        # So a frequency loop will not be too large.
-
-        weight_local = weight.local_array
 
-        for lfi, gfi in weight.enumerate(axis=0):
+        weight_local = data.weight[:].local_array
 
-            # MPIArray takes the local index, returns a local np.ndarray
+        for i in range(weight_local.shape[0]):
             # Find values equal to zero to preserve them in final weights
-            zeromask = weight_local[lfi] == 0.0
-            # Median filter. Mode='nearest' to prevent steps close to
-            # the end from being washed
-            weight_local[lfi] = median_filter(
-                weight_local[lfi], size=(1, self.kernel_size), mode="nearest"
+            zeromask = weight_local[i] == 0.0
+
+            weight_local[i] = weighted_median.moving_weighted_median(
+                data=weight_local[i],
+                weights=np.ones_like(weight_local[i]),
+                size=(1, self.kernel_size),
```

Comment:
Not sure if there's a better/more efficient way to make the weights. Maybe setting dtype to int instead of float